### ISSUE TITLE
Windows ODBC 

### ISSUE THREAD

**Issue by bucweat** [duckdb/duckdb#3398]
#### What happens?
Compiled DuckDb from source on 64 bit Windows 11 using Visual Studio 2019. I basically followed the build process identified in the windows.yml used by github CI. I did this for both 32 and 64 bit. I then ran unittest.exe and both 32/64 bit passed 100%. I then installed the 32/64 bit ODBC drivers located in .../tools/odbc/bin (note that I had to copy the .exe and .dll files up one level to odbc in order to avoid error about not being able to find duckdb_odbc_setup.dll). Install was successful. Poking at the windows registry shows that both drivers are installed. I then attempted to query DuckDB using ADO which failed with the following error:

`[Microsoft][ODBC Driver Manager] The driver is incapable of supporting the current environment attributes.`

I discussed this on the DuckDB #general channel on Discord initially and in much more detail with @tiagokepe. I promised him that I would submit an issue with details and also provide a set of scripts that he could use to illustrate ADO to touch the ODBC driver as a test case. The scripts also include support for SQLiteODBC to illustrate how things can be done in SQLite.

Discussions with @tiagokepe indicated that only DSN was currently supported. So it is possible that I'm not generating the connection string correctly for DSN. However, one of my goals here is to facilitate adding support for "Driver=" connection string where the database file name can be provided in the connection string via the "Database=" attribute like SQLiteODBC does. 

Whats attached:
* odbcTest.zip - contains script files and README detailing how to use the scripts to reproduce issue.
* duckdbodbc.txt - trace created with `ODBC Data Source Administrator` with failed attempt to access DuckDB using ADO
* sqliteodbc.txt -  trace created with `ODBC Data Source Administrator` with successful attempt to access SQLiteODBC using ADO

#### To Reproduce
See the attached .zip file with scripts that will illustrate the problem.

#### Environment (please complete the following information):
 - OS: Windows 11 21H2 64 bit
 - DuckDB Version: master cloned on 7 April 2022
 - DuckDB Client: ODBC

#### Before Submitting

- [X] **Have you tried this on the latest `master` branch?**
- [ ] Not a Python or R issue.
- * **Other Platforms**: You can find binaries [here](https://github.com/duckdb/duckdb/releases/tag/master-builds) or compile from source.
    - I have not tried the ODBC binaries from the release page. I compiled from source.
- [X] **Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?** Yes the attached .zip file contains a pretty detailed set of scripts to reproduce the issue. 

[odbcTest.zip](https://github.com/duckdb/duckdb/files/8448905/odbcTest.zip)
[duckdbodbc.txt](https://github.com/duckdb/duckdb/files/8448908/duckdbodbc.txt)
[sqliteodbc.txt](https://github.com/duckdb/duckdb/files/8448910/sqliteodbc.txt)

**Comment by tiagokepe** [comment]
> #### What happens?
> Compiled DuckDb from source on 64 bit Windows 11 using Visual Studio 2019. I basically followed the build process identified in the windows.yml used by github CI. I did this for both 32 and 64 bit. I then ran unittest.exe and both 32/64 bit passed 100%. I then installed the 32/64 bit ODBC drivers located in .../tools/odbc/bin (note that I had to copy the .exe and .dll files up one level to odbc in order to avoid error about not being able to find duckdb_odbc_setup.dll). Install was successful. 

I couldn't reproduce this situation, I tested and used the installer in a similar setting (i.e., launching it inside the directory `duckd/build/release/tools/odbc/bin/Release/`).
The only requirement to run the odbc_install.exe is to launch it from the same directory where the .dll files are placed.

> Poking at the windows registry shows that both drivers are installed. I then attempted to query DuckDB using ADO which failed with the following error:
> 
> `[Microsoft][ODBC Driver Manager] The driver is incapable of supporting the current environment attributes.`
> 

I made some adjustments (see PR #3401) and then I could run the .vbs script on my environment. It seems to work ;).

> I discussed this on the DuckDB #general channel on Discord initially and in much more detail with @tiagokepe. I promised him that I would submit an issue with details and also provide a set of scripts that he could use to illustrate ADO to touch the ODBC driver as a test case. The scripts also include support for SQLiteODBC to illustrate how things can be done in SQLite.
> 
> Discussions with @tiagokepe indicated that only DSN was currently supported. So it is possible that I'm not generating the connection string correctly for DSN. However, one of my goals here is to facilitate adding support for "Driver=" connection string where the database file name can be provided in the connection string via the "Database=" attribute like SQLiteODBC does.

I did some tests and it's possible to use the `database` key inside the connection string, like **"Driver=DuckDB Driver;Database=my_file.db"**. This connection string worked with the ISQL Linux command.
The issue was not related to the SQLDriverConnect function, instead it was linked to environment and connection attributes (see PR #3401).

> 
> Whats attached:
> 
> * odbcTest.zip - contains script files and README detailing how to use the scripts to reproduce issue.
> * duckdbodbc.txt - trace created with `ODBC Data Source Administrator` with failed attempt to access DuckDB using ADO
> * sqliteodbc.txt -  trace created with `ODBC Data Source Administrator` with successful attempt to access SQLiteODBC using ADO

Thanks for the files, they helped to track down the issue.
> 
> #### To Reproduce
> See the attached .zip file with scripts that will illustrate the problem.
> 
> #### Environment (please complete the following information):
> * OS: Windows 11 21H2 64 bit
> * DuckDB Version: master cloned on 7 April 2022
> * DuckDB Client: ODBC
> 
> #### Before Submitting
> * [x]  **Have you tried this on the latest `master` branch?**
> * [ ]  Not a Python or R issue.
> * * **Other Platforms**: You can find binaries [here](https://github.com/duckdb/duckdb/releases/tag/master-builds) or compile from source.
>     
>     * I have not tried the ODBC binaries from the release page. I compiled from source.
> * [x]  **Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?** Yes the attached .zip file contains a pretty detailed set of scripts to reproduce the issue.
> 
> [odbcTest.zip](https://github.com/duckdb/duckdb/files/8448905/odbcTest.zip) [duckdbodbc.txt](https://github.com/duckdb/duckdb/files/8448908/duckdbodbc.txt) [sqliteodbc.txt](https://github.com/duckdb/duckdb/files/8448910/sqliteodbc.txt)

**Comment by bucweat** [comment]
> > Compiled DuckDb from source on 64 bit Windows 11 using Visual Studio 2019. I basically followed the build process identified in the windows.yml used by github CI. I did this for both 32 and 64 bit. I then ran unittest.exe and both 32/64 bit passed 100%. I then installed the 32/64 bit ODBC drivers located in .../tools/odbc/bin (note that I had to copy the .exe and .dll files up one level to odbc in order to avoid error about not being able to find duckdb_odbc_setup.dll). Install was successful.
> 
> I couldn't reproduce this situation, I tested and used the installer in a similar setting (i.e., launching it inside the directory `duckd/build/release/tools/odbc/bin/Release/`). The only requirement to run the odbc_install.exe is to launch it from the same directory where the .dll files are placed.

![image](https://user-images.githubusercontent.com/3979338/162574627-45ef6ac0-c4e7-4b37-9101-0fb258a4690a.png)

Note that I'm in bin yet it is looking for .dll one directory up. Yes the subject .dll is in bin. If I copy 2 x .dll and .exe up one level then it works. Yeah I don't get it either...it's probably has something to do with how I'm building in windows. 

My process is open a visual studio 2019 dev cmd for arch I want to build (x86 or x64), cd to duckdb, create a build folder (build32 or build64), cd to build folder for arch I'm building and then run the following:

```
    cmake -S ..\ -B . -G "NMake Makefiles" ^
        -DCMAKE_BUILD_TYPE=Release  ^
        -DBUILD_PARQUET_EXTENSION=1  ^
        -DBUILD_FTS_EXTENSION=1  ^
        -DBUILD_ODBC_DRIVER=1  ^
        -DDISABLE_UNITY=1
        
    cmake --build . --config Release
```

Actually that's all is in a batch script. Not saying its right but it's what I'm currently doing as I could find no directions for building on Windows and it otherwise seems to work as all unit tests pass and duckdb.exe seems to work fine with all the sql files I've throw at it so far.

**Comment by bucweat** [comment]
@tiagokepe I see the commit to master happened yesterday...I'll give that a try and let you know how it goes here. Thanks :-)

**Comment by tiagokepe** [comment]
> @tiagokepe I see the commit to master happened yesterday...I'll give that a try and let you know how it goes here. Thanks :-)

Great!

About the installer, we currently support the build generator for "Visual Studio" that sets the installer and files path (.dll / .exe) properly.
To avoid setup issues (like that with ````nmake````), you can download the installer from the [Latest Release](https://github.com/duckdb/duckdb/releases/download/v0.3.3/duckdb_odbc-windows-amd64.zip).

**Comment by bucweat** [comment]
Thanks I’m on a long weekend away so won’t get to look until Monday. WRT installer one of my requirements is I need to compile from source….no binaries. I know how to work around the compiled installer so that is fine. I would like to understand what is causing installer .exe to not to look in directory it lives in for .dll files…that just isn’t right 🙂

**Comment by tiagokepe** [comment]
> Thanks I’m on a long weekend away so won’t get to look until Monday. WRT installer one of my requirements is I need to compile from source….no binaries. I know how to work around the compiled installer so that is fine. I would like to understand what is causing installer .exe to not to look in directory it lives in for .dll files…that just isn’t right 🙂

Hey @bucweat , I detected a legacy code that manipulates the driver file path.
I removed that code in my fork (https://github.com/tiagokepe/duckdb/commit/cd8eb7ebc09014e03e97e62d1886a138d571353c).
Could you test if my changes fixed the installer path problem in your environment?
You can checkout the master branch from my fork (```` git checkout https://github.com/tiagokepe/duckdb.git````) or apply that changes locally.

**Comment by bucweat** [comment]
Wow copy will do when I get home Monday 😊

**Comment by bucweat** [comment]
Hi @tiagokepe I was successful updating to latest duckdb/master and also rolling in your change to install.c...git novice here so it took a little doing... 

Here is status:

* rebuilt 32/64 bit duckdb successfully. I manually deleted build[32]64]/tools/odbc/bin to get all the .dll/.exe in there to rebuild
* uninstalled/installed 32/64 bit from the build[32]64]/tools/odbc/bin directory successfully (looks like your change to install.c did the trick to install from bin)
* ran my test script (attached above) using connection string `Driver=DuckDB Driver;Database=.\test.duckdb;`
    * ADO connection opened successfully 
        * test.duckdb file created, 12 kB in size
        * test.duckdb.wal file created, 0 kB in size
    * the first SQL in the test script `DROP TABLE if exists 'weather';` causes the VBScript to die without an error. 
    * I commented out the DROP to see if next SQL insert would work, and it fails the same way
        * attached odbc.txt trace file from ODBC Data Source Administrator app trace feature.
    * the resulting test.duckdb opens via the command line app, so the db is at least well formed, but no contents.

So definitely a few steps forward for me, but not sure about the script dying without error part...does the script run successfully on your setup?

[odbc.txt](https://github.com/duckdb/duckdb/files/8503723/odbc.txt)

**Comment by tiagokepe** [comment]
> Hi @tiagokepe I was successful updating to latest duckdb/master and also rolling in your change to install.c...git novice here so it took a little doing...
> 
> Here is status:
> 
> * rebuilt 32/64 bit duckdb successfully. I manually deleted build[32]64]/tools/odbc/bin to get all the .dll/.exe in there to rebuild
> * uninstalled/installed 32/64 bit from the build[32]64]/tools/odbc/bin directory successfully (looks like your change to install.c did the trick to install from bin)
> * ran my test script (attached above) using connection string `Driver=DuckDB Driver;Database=.\test.duckdb;`
>   
>   * ADO connection opened successfully
>     
>     * test.duckdb file created, 12 kB in size
>     * test.duckdb.wal file created, 0 kB in size
>   * the first SQL in the test script `DROP TABLE if exists 'weather';` causes the VBScript to die without an error.
>   * I commented out the DROP to see if next SQL insert would work, and it fails the same way
>     
>     * attached odbc.txt trace file from ODBC Data Source Administrator app trace feature.
>   * the resulting test.duckdb opens via the command line app, so the db is at least well formed, but no contents.
> 
> So definitely a few steps forward for me, but not sure about the script dying without error part...does the script run successfully on your setup?
> 
> [odbc.txt](https://github.com/duckdb/duckdb/files/8503723/odbc.txt)

Hi @bucweat, great that you could advance with the installer.
I ran your script successfully, but I will give a double check.
In addition, I submitted a new PR #3446 with some adjustments for PowerBI working (perhaps that could help in some way).

**Comment by tiagokepe** [comment]
Hey @bucweat,
I made some tweaks (see PR #3467), then your script throws the following syntax error:
````
Parser Error: syntax error at or near "'weather'"
LINE 1: DROP TABLE if exists 'weather';
````

**Comment by bucweat** [comment]
Hi @tiagokepe,

Hummm...interesting. Assuming you can open command line and execute that query successfully? I'm assuming the SQL is valid for duckdb (see https://duckdb.org/docs/sql/statements/drop) It does work in SQLite.

I think you are getting farther along in the script process than I am as it seems you get a successful connection  to open and are to the point of executing SQL. I don't get past the attempting to open a connection.... Assuming you are running in a standard CMD shell without admin privileges. I did try with admin CMD with same results as without. 

I also made sure that the new driver is getting installed but running /UnInstall and then running the script which does report that it cannot find DuckDB ODBC driver. I then installed with /Install and reran the script and it does find the driver. 

Here is what I did: I updated git to latest (commit `f63df22eb8e75f682a6a05bf53627a5ad38c8398 ` dated 20 April 2022), recompiled 64 bit, reinstalled, and ran my test script `runOdbcTest.bat 64` and here is the output. Note that I have SQLite ODBC driver installed so that you can see what the expected output should look like from DuckDB. Also note that the queries defined in the script do work with the command line duckdb.exe.

```
***********************
***** runOdbcTest *****
***********************
Running tests as 64 bit process...
strFolder [M:\Charlie\odbcTest_DUCKDB]
class_initialize
classOdbcTests configured to run 64bit
architecture reports:
    SYSTEM AMD64
    PROCESS AMD64
ODBC Providers: (took 0 seconds)
    SQL Server - Installed
    SQLite3 ODBC Driver - Installed
    DuckDB Driver - Installed


****************************************************************************
simpleSQLiteTest with file test.sqlite3
SQL3 DRIVER=SQLite3 ODBC Driver;Database=test.sqlite3;

QUERY  DROP TABLE if exists 'weather';
return -1
ERROR  query did not return an open recordset|

QUERY  CREATE TABLE weather (     city           VARCHAR,     temp_lo        INTEGER,    temp_hi        INTEGER,    prcp           REAL,     date           DATE );
return -1
ERROR  query did not return an open recordset|

QUERY  PRAGMA table_info('weather');
return 5 rows
cid(adInteger),name(adVarWChar),type(adVarWChar),notnull(adInteger),dflt_value(adVarWChar),pk(adInteger)
0,"city","VARCHAR",0,Null,0
1,"temp_lo","INTEGER",0,Null,0
2,"temp_hi","INTEGER",0,Null,0
3,"prcp","REAL",0,Null,0
4,"date","DATE",0,Null,0

QUERY  INSERT INTO weather VALUES ('San Francisco', 46, 50, 0.25, '1994-11-27');
return -1
ERROR  query did not return an open recordset|

QUERY  select * from weather;
return 1 rows
city(adVarWChar),temp_lo(adInteger),temp_hi(adInteger),prcp(adDouble),date(adDBDate)
"San Francisco",46,50,0.25,11/27/1994

****************************************************************************
selecting file via connection string using 'Driver=' and 'Database=' like SQLiteODBC
DUCKDB Driver=DuckDB Driver;Database=test.duckdb;

(base) M:\Charlie\odbcTest_DUCKDB>
```

For SQLite the `ERROR  query did not return an open recordset|` is not really an error...it just indicates that a recordset was not returned when the query was executed which is perfectly fine for cases where the SQL is not expected to return results (queries with DROP/CREATE/INSERT/etc).

Digging into the script a little farther, the open call to the connection here fails causing the script to crash without reporting any errors so no SQL is getting run here currently.

The `test.duckdb` file does get created, and is a valid database file (e.g. opens successfully with command line app), but contains no tables.

**Comment by tiagokepe** [comment]
The changes in PR #3504 might fix some points in this issue.

**Comment by bucweat** [comment]
Hi @tiagokepe assuming I'm reading the commit tea leaves correctly this has not been merged into repo yet? Are these changes in HEAD of your fork? I'm assuming they are given you issued pull request from there but still coming up to speed on git...

**Comment by tiagokepe** [comment]
> Hi @tiagokepe assuming I'm reading the commit tea leaves correctly this has not been merged into repo yet? Are these changes in HEAD of your fork? I'm assuming they are given you issued pull request from there but still coming up to speed on git...

It seems that the PRs were merged into the master.

**Comment by bucweat** [comment]
Yeah was happily building and ran into an issue caused by one of the merged PRs. I might have jumped the gun a little early on the pull...pulled again with some updates and attempting rebuild.

**Comment by tiagokepe** [comment]
> Well had some weird linking errors so had delete build folders and start from scratch. 64 bit is built, installed, and it is now working with the Database= connection string. Running my test VBScript, the DROP/CREATE/INSERT all work (removing the quotes around weather in the DROP as you mentioned) and a table is built in test.duckdb and when I look in duckdb.exe table looks reasonable
> 
> ```
> duckdb test.duckdb
> v0.3.5-dev164 e4ba94a4f
> Enter ".help" for usage hints.
> D select * from weather;
> ┌───────────────┬─────────┬─────────┬──────┬────────────┐
> │     city      │ temp_lo │ temp_hi │ prcp │    date    │
> ├───────────────┼─────────┼─────────┼──────┼────────────┤
> │ San Francisco │ 46      │ 50      │ 0.25 │ 1994-11-27 │
> │ San Francisco │ 46      │ 50      │ 0.25 │ 1994-11-27 │
> └───────────────┴─────────┴─────────┴──────┴────────────┘
> D .schema weather
> CREATE TABLE weather(city VARCHAR, temp_lo INTEGER, temp_hi INTEGER, prcp FLOAT, date DATE);;
> ```

Very good that we are making progress!

**Comment by tiagokepe** [comment]
> The OLE DB error I think has to do with the return type for the date column adDBDate type #133. Still looking at if VBScript can handle this type from ADO.
> 
> It would seem that the single INSERT query in the script is getting execute twice, hence the two rows in the table in the previous post. Added another INSERT with different city...results in 4 rows in the table.

This is weird, I need to verify.

> 
> 64 bit driver compiled fine, but still seeing `duckdb_odbc.dll` link issue with 32 bit build even after doing a clean build from scratch. Figure I'd share that here first since it is ODBC driver and you might recognize:
> 
> ```
> .
> .
> .
> [ 89%] Linking CXX shared library bin\duckdb_odbc.dll
> LINK: command "C:\PROGRA~2\MICROS~2\2019\PROFES~1\VC\Tools\MSVC\1429~1.301\bin\Hostx86\x86\link.exe /nologo @CMakeFiles\duckdb_odbc.dir\objects1.rsp /out:bin\duckdb_odbc.dll /implib:duckdb_odbc.lib /pdb:M:\Apps\github\duckdb\build32\tools\odbc\bin\duckdb_odbc.pdb /dll /version:0.0 /machine:X86 /INCREMENTAL:NO /DEF:..\..\..\tools\odbc\duckdb_odbc.def odbccp32.lib legacy_stdio_definitions.lib ..\..\src\duckdb_static.lib ws2_32.lib ..\..\third_party\fmt\duckdb_fmt.lib ..\..\third_party\libpg_query\duckdb_pg_query.lib ..\..\third_party\re2\duckdb_re2.lib ..\..\third_party\miniz\duckdb_miniz.lib ..\..\third_party\utf8proc\duckdb_utf8proc.lib ..\..\third_party\hyperloglog\duckdb_hyperloglog.lib ..\..\third_party\fastpforlib\duckdb_fastpforlib.lib ..\..\extension\parquet\parquet_extension.lib ..\..\extension\fts\fts_extension.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:bin\duckdb_odbc.dll.manifest" failed (exit code 1120) with the following output:
>    Creating library duckdb_odbc.lib and object duckdb_odbc.exp
> statement.cpp.obj : error LNK2019: unresolved external symbol _SQLColAttribute@28 referenced in function _SQLColAttributes@28
>   Hint on symbols that are defined and could potentially match:
>     "short __stdcall SQLColAttribute(void *,unsigned short,unsigned short,void *,short,short *,long *)" (?SQLColAttribute@@YGFPAXGG0FPAFPAJ@Z)
> bin\duckdb_odbc.dll : fatal error LNK1120: 1 unresolved externals
> NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe"' : return code '0xffffffff'
> .
> .
> .
> ```
> 
> Up until this last pull the 32 bit code build fine. Hopefully you'll see something there that makes sense to you.

I see the problem with the new function SQLColAttributes(). It should export as external symbol for Windows by adding it at the file `tool/odbc/duckdb_odbc.def`. I made it in my [fork](https://github.com/tiagokepe/duckdb/commit/a9fdf4490dc28740cdcff8d3f5890aadabc793b5) case you want to pull.

**Comment by bucweat** [comment]
Strange as it may seem, that did not fix it. I completely wiped my build32 directory and built it from scratch, and no change. Error is exactly the same as before. Weird that it builds just fine 64 bit but not 32 bit. 

Just checked the .github\workflows\Windows.yml and the 32 bit build does not include `-DBUILD_ODBC_DRIVER=1 ` so if I'm reading that right the 32 bit ODBC driver does not get built by the CI?

**Comment by tiagokepe** [comment]
> Just checked the .github\workflows\Windows.yml and the 32 bit build does not include `-DBUILD_ODBC_DRIVER=1 ` so if I'm reading that right the 32 bit ODBC driver does not get built by the CI?

You are right, we skipped Windows 32-bits CI because we only build, test, and release ODBC assets on Windows 64-bits for now.

**Comment by tiagokepe** [comment]
> It would seem that the single INSERT query in the script is getting execute twice, hence the two rows in the table in the previous post. Added another INSERT with different city...results in 4 rows in the table.

To isolate the problem, I ran the script with only the "DUCKDB" using the `database` parameter and I figured out in the odbc trace file that the `INSERT` query was executed twice through two distinct calling of `SQLExecDirect`.

Could you provide the ODBC trace files for a single execution with sqlite3 and another with duckdb?

**Comment by bucweat** [comment]
Here ya go...once with just sqlite, once with just duckdb in separate files. Looks like duckdb executes the drop/create only once, but both the insert/select each twice. 

Console output here, trace in attached files. Hope this helps :-)

```
****************************************************************************
simpleSQLiteTest with file test.sqlite3
SQL3 DRIVER=SQLite3 ODBC Driver;Database=test.sqlite3;

QUERY  DROP TABLE if exists weather;
return -1
ERROR  query did not return an open recordset|

QUERY  CREATE TABLE weather (     city           VARCHAR,     temp_lo        INTEGER,    temp_hi        INTEGER,    prcp           REAL,     date           DATE );
return -1
ERROR  query did not return an open recordset|

QUERY  INSERT INTO weather VALUES ('San Francisco', 46, 50, 0.25, '1994-11-27');
return -1
ERROR  query did not return an open recordset|

QUERY  select * from weather;
return 1 rows
city(adVarWChar),temp_lo(adInteger),temp_hi(adInteger),prcp(adDouble),date(adDBDate)
"San Francisco",46,50,0.25,Date

class_terminate
```

```
selecting file via connection string using 'Driver=' and 'Database=' like SQLiteODBC
DUCKDB Driver=DuckDB Driver;Database=test.duckdb;

QUERY  DROP TABLE if exists weather;
return -1
ERROR  recordset contains no records|

QUERY  CREATE TABLE weather (     city           VARCHAR,     temp_lo        INTEGER,    temp_hi        INTEGER,    prcp           REAL,     date           DATE );
return -1
ERROR  recordset contains no records|

QUERY  INSERT INTO weather VALUES ('San Francisco', 46, 50, 0.25, '1994-11-27');
return 1 rows
Count(adBigInt)
0

M:\Charlie\odbcTest_DUCKDB\odbcTest.wsf(481, 39) Microsoft OLE DB Provider for ODBC Drivers: Multiple-step OLE DB operation generated errors. Check each OLE DB status value, if available. No work was done.

class_terminate
```

[sqlite_odbc.txt](https://github.com/duckdb/duckdb/files/8574081/sqlite_odbc.txt)
[duckdb_odbc.txt](https://github.com/duckdb/duckdb/files/8574085/duckdb_odbc.txt)

**Comment by tiagokepe** [comment]
Hey @bucweat,

It seems that PR #3646 has fixed this issue.
Do you mind if I add your VBscript (with some simplifications) into the ODBC test suite?

**Comment by bucweat** [comment]
Nope have at it.
